plugins/http_plugin/include/eosio/http_plugin/beast_http_session.hpp

```diff
@@ -75,6 +78,26 @@ bool allow_host(const http::request<http::string_body>& req, T& session,
    return true;
 }
 
+// T can be request or response or anything serializable to boost iostreams
+template<typename T>
+std::string to_log_string(const T& req, size_t max_size = 1024) {
+   assert( max_size > 4 );
+   std::string buffer( max_size, '\0' );
+   {
+      boost::iostreams::array_sink sink( buffer.data(), buffer.size() );
+      boost::iostreams::stream stream( sink );
+      stream << req;
+   }
+   buffer.resize( std::strlen( buffer.data() ) );
+   if( buffer.size() == max_size ) {
```

Comment:
Should `==` be `>=` and truncate buffer between max_size .. buffer.size()?

Reply:
`buffer.size()` can't be greater than max_size.

Reply:
Oh, didn't notice buffer was initialized to max_size.